src/main/kotlin/com/kostas/kafkaStreams/rest/Controller.kt

```diff
@@ -0,0 +1,30 @@
+package com.kostas.kafkaStreams.rest
+
+import arrow.core.getOrHandle
+import com.kostas.kafkaStreams.ListingService
+import com.kostas.kafkaStreams.metadata.ApiClientFactory
+import com.kostas.kafkaStreams.model.ListingSummary
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import javax.ws.rs.QueryParam
+
+@RestController
+@RequestMapping("/listings")
+class Controller(
+    private val listingService: ListingService,
+    private val apiClient: ApiClientFactory
+): ListingApi {
```

Comment:
You could make this more "REST" by having one endpoint `GET /listings` that has two optional query params `postCode` & `propertyType`